### PR TITLE
Incude Company in Contact

### PR DESCRIPTION
**Pull request by sun-ruiheng**
Screenshot 2024-03-14 at 3 04 04 PM

### CODE REVIEW

**Comment by guanquann** [comment]
- When I entered `add n/James Ho p/22224444 e/jamesho@example.com a/123, Clementi Rd, 1234665 t/friend t/colleague` (without the company), it did not show any change. 

- And it looks like you need to add the company prefix tag in `EditCommandParser.java`, as I can't seem to edit a person's company using `edit 7 c/ nusfishery `

**Comment by sun-ruiheng** [comment]
> * When I entered `add n/James Ho p/22224444 e/jamesho@example.com a/123, Clementi Rd, 1234665 t/friend t/colleague` (without the company), it did not show any change.
> * And it looks like you need to add the company prefix tag in `EditCommandParser.java`, as I can't seem to edit a person's company using `edit 7 c/ nusfishery `

Thanks Guan Quan,
The first bug pointed out has been fixed.
The second point is the way because Company is a compulsory field in v1.2.

**Review by gavingoh99 — COMMENTED** [review]
EditCommandParser seems to flag out empty prefixes before invalid index (> number of contacts):
edit [invalid index] n/ -> returns Name cannot be empty rather than displaying the index > number of contacts error message

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -101,11 +104,12 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
         Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
         Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
+        Company updatedCompany = editPersonDescriptor.getCompany().orElse(personToEdit.getCompany());
         Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
         ArrayList<Order> updatedOrders = personToEdit.getOrders();
```
Comment (gavingoh99) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:110]:
```suggestion
    boolean updatedIsFavourite = personToEdit.getFavourite();
```

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -101,11 +104,12 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
...
 
-        return new Person(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedTags,
-                updatedOrders);
+        return new Person(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedCompany,
+                updatedTags, updatedOrders);
```
Comment (gavingoh99) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:112]:
```suggestion
        return new Person(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedCompany, updatedIsFavourite, updatedTags, updatedOrders);
```

**Review by gavingoh99 — CHANGES_REQUESTED** [review]
Currently EditCommand resets the favourite field of a contact when edited

I've provided suggestions to preserve the favourite field - do have a look!

**Review by gavingoh99 — APPROVED** [review]
LGTM! Nice handling of the issues raised.

**Pull request merged**